pkg/bloomgateway/client_pool.go

```diff
@@ -80,35 +93,47 @@ func (p *JumpHashClientPool) Addr(key string) (string, error) {
 	return addr.String(), nil
 }
 
-func (p *JumpHashClientPool) Start() {
-	ctx := context.Background()
-	_ = services.StartAndAwaitRunning(ctx, p.Pool)
+func (p *JumpHashClientPool) updateLoop(_ context.Context) error {
+	servers := p.provider.Addresses()
+	// ServerList deterministically maps keys to _index_ of the server list.
+	// Since DNS returns records in different order each time, we sort to
+	// guarantee best possible match between nodes.
+	sort.Strings(servers)
```

Comment:
Lexicographic sort isn't necessary here (and is incorrect for jumphash); jumphash needs natural sorting of servers so that `z11` sorts after `z2`. The call to `SetServers` below will do a natural sort at the top of its implementation.